src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs

```diff
@@ -147,14 +148,16 @@ public int CalculateOffsetToMappedFieldDataStream()
 
         internal int ComputeOffsetToDebugDirectory()
         {
-            Debug.Assert(MetadataSize % 4 == 0);
-            Debug.Assert(ResourceDataSize % 4 == 0);
+            Debug.Assert(MetadataSize % MetadataSizes.StreamAlignment == 0);
 
-            return
+            int offset =
                 ComputeOffsetToMetadata() +
                 MetadataSize +
-                ResourceDataSize +
+                BitArithmetic.Align(ResourceDataSize, ManagedResourcesDataAlignment) +
                 StrongNameSignatureSize;
+
+            Debug.Assert(offset % MetadataSizes.StreamAlignment == 0);
```

Comment:
`StrongNameSignatureSize` is an arbitrary number passed in by the user code. I do not think that this assert holds - it should be possible to write a simple test that hits it.

Reply:
Yes thanks - removed that Assert. A quick test showed that passing in a non-aligned value to the `ManagedPEBuilder` ctor would trigger this assert, and there is no doc that I can find that says the value needs to be aligned. The only check we have is for negative.